--- NVIDIA-Linux-x86-340.107/kernel/nv-drm.c~	2019-12-01 19:14:03.000000000 +0100
+++ NVIDIA-Linux-x86-340.107/kernel/nv-drm.c	2019-12-01 19:15:23.503016305 +0100
@@ -149,7 +149,11 @@
 #if defined(DRIVER_LEGACY)
     .driver_features = DRIVER_GEM | DRIVER_PRIME | DRIVER_LEGACY,
 #else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0)
     .driver_features = DRIVER_GEM | DRIVER_PRIME,
+#else
+    .driver_features = DRIVER_GEM,
+#endif
 #endif
     .load = nv_drm_load,
     .unload = nv_drm_unload,
@@ -161,7 +165,9 @@
     .gem_free_object = nv_gem_free,
 
     .prime_handle_to_fd = drm_gem_prime_handle_to_fd,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0)
     .gem_prime_export = drm_gem_prime_export,
+#endif
     .gem_prime_get_sg_table = nv_gem_prime_get_sg_table,
     .gem_prime_vmap = nv_gem_prime_vmap,
     .gem_prime_vunmap = nv_gem_prime_vunmap,
