--- NVIDIA-Linux-x86-1.0-6106-pkg1/usr/src/nv/Makefile.kbuild.orig	2004-06-23 17:51:51.000000000 +0200
+++ NVIDIA-Linux-x86-1.0-6106-pkg1/usr/src/nv/Makefile.kbuild	2004-07-04 03:15:41.290378952 +0200
@@ -73,7 +73,7 @@
 #
 
 EXTRA_CFLAGS += -I$(src)
-EXTRA_CFLAGS += -Wall -Wimplicit -Wreturn-type -Wswitch -Wformat -Wchar-subscripts -Wparentheses -Wpointer-arith  -Wno-multichar  -Werror -O -fno-common -MD $(DEFINES) $(INCLUDES) -Wno-cast-qual -Wno-error
+EXTRA_CFLAGS += -Wall -Wimplicit -Wreturn-type -Wswitch -Wformat -Wchar-subscripts -Wparentheses -Wno-multichar  -Werror -O -fno-common -MD $(DEFINES) $(INCLUDES) -Wno-cast-qual -Wno-error
 
 #
 # We rely on these two definitions below; if they aren't set, we set them to
